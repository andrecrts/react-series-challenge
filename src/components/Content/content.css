.portal {
  width: 500px;
  height: 600px;
  margin:auto;
  background-color:gray;
  text-align:center;
}

.portal > img {
  width:100%;
  height:500px;
}

.content {
  position:absolute;
  overflow:hidden;
  z-index:-1;
}

.btn-close {
  width: 100%;
  height: 40px;
  background-color:green;
  font-size: 20px;
}
