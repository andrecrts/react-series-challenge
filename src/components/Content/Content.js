import React, { Component } from 'react'
import Item from '../Item'
import { createPortal } from 'react-dom'
import './content.css'

const ContentList = ({data = {}, onClickFav, favorites, filter, onClick}) => {

  const isEmptyObject =  (obj) => {
    for(var prop in obj) {
        return false
    }
    return true
  }

  if(data.hasOwnProperty('data')){
    return data.data.map((item, index) => (
      <Item key={index}
        item={item}
        onClickFav={onClickFav}
        isFav={favorites[item.id] ? true : false}
        onClick={onClick}
      />
    ))
  } else {
  return !isEmptyObject(data) ? Object.keys(data).map((key, index) => {
    return data[key].title.search(new RegExp(filter, 'i')) > -1 ?
      (
          <Item key={index}
            item={data[key]}
            onClickFav={onClickFav}
            isFav={favorites[key] ? true : false}
            onClick={onClick}
          />
      ) : null
  }) : <div> No Favorites </div>
  }

}

const PortalContainer = ({children}) => {
  const nodeToAppend = document.getElementsByTagName('body')[0]

  return createPortal(children, nodeToAppend)
}

const Portal = ({open, onClick, item}) => {
    return open ? (
      <PortalContainer>
        <div className="portal">
          <button onClick={onClick} className="btn-close"> Close </button>
          <img src={item.images.original.webp} alt={item.title}/>
          <p> {item.title} </p>
        </div>
        </PortalContainer>
    )
  : null
}

class Content extends Component {

  constructor(props) {
    super(props)
    this.state = { open: false , item: ''}
    this.togglePortal = this.togglePortal.bind(this)
  }

  togglePortal(item = null) {
    const { open } = this.state
    if(open){
      this.setState({open: !open})
    } else {
      this.setState({ open: !open, item})
    }
  }

  render() {
    const {data = {}, isLoading = false, onClickFav, favorites, filter} = this.props
    if(isLoading){
      return(
        <div> loading ... </div>
      )
    }

    return(
      <div className="content">
        <ContentList
          data={data}
          favorites={favorites}
          onClickFav={onClickFav}
          filter={filter}
          onClick={this.togglePortal}
        />
        <Portal
          open={this.state.open}
          onClick={this.togglePortal}
          item = {this.state.item}
        />
      </div>
    )

  }
}

export default Content
