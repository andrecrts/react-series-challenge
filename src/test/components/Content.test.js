import React from 'react';
import Content from '../../components/Content'
import renderer from 'react-test-renderer';

test('Home snapshot', () => {
  const data= {
          data1: {
            title:'title1', images:{original:{webp:'srcImage'}}
          }}
  const favorites = {}
  const component = renderer.create(
    <Content
      data={data}
      favorites= {favorites}
    />
  );
  let componentJSON = component.toJSON();
  expect(componentJSON).toMatchSnapshot();
})

