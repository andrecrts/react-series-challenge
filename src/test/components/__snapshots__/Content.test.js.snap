// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<div
  className="sc-ifAKCX iySRWN"
>
  <div
    className="sc-bdVaJa jFnFkT"
  >
    <img
      alt="Favorite"
      className="sc-bwzfXH caOhOO"
      onClick={[Function]}
      src="no-fav.svg"
    />
    <img
      alt="title1"
      className="sc-htpNat kewJQA"
      onClick={[Function]}
      src="srcImage"
    />
  </div>
</div>
`;
